pkg/internal/kube/store.go

```diff
@@ -35,10 +49,12 @@ type Store struct {
 	namespaces map[uint32]*container.Info
 
 	// container ID to pod matcher
-	podsByContainer map[string]*informer.ObjectMeta
+	podsByContainer   map[string]*informer.ObjectMeta
+	containersByOwner map[string][]*informer.ContainerInfo
 
 	// ip to generic IP info (Node, Service, *including* Pods)
-	ipInfos map[string]*informer.ObjectMeta
+	ipInfos             map[string]*informer.ObjectMeta
+	otelServiceInfoByIP map[string]OTelServiceNamePair
```

Comment:
this won't be active in non-kubernetes environments. Is it ok?

Reply:
It should be fine, because we fetch the service name from env vars too, so non kube shouldn't be affected. Actually, I meant to ask, is there a way I can tell early in Beyla's startup if k8s is enabled and if we have our metadata service on?

Reply:
We use the `IsKubeEnabled` method in `MetadataProvider` to know that during startup.

For the metadata service, I'm working on a PR that would let Beyla know when all the kubernetes entities have been locally synced from the Kube metadata. So I guess that it would let knowing certainly that the metadata service is on.